apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php

```diff
@@ -60,6 +61,21 @@ function register_blocks() {
 			'render_callback' => 'render_template_part_block',
 		) );
 	}
+
+	/**
+	 * This will set the `wp_template` post type to `public` to support
+	 * the core search endpoint, which looks for it.
+	 *
+	 * @return void
+	 */
+	function allow_searching_for_templates() {
+		$post_type = get_post_type_object( 'wp_template' );
+		if ( ! is_a( $post_type, 'WP_Post_Type' ) ) {
+			return;
+		}
+		// setting this to `public` will allow it to be found in the search endpoint
+		$post_type->public = true;
```

Comment:
Interesting!
Code works as expected, but I have to ask: is this kind of mutation an established enough pattern, or is it something that might end up frowned upon when/if we push this into Core?

Reply:
There are sometimes workarounds somewhat like this in core, but usually what they expose is a need for changes elsewhere, which would generally happen as part of a merge strategy.